nucleus/core/bootstrap/src/main/java/fish/payara/boot/runtime/BootCommands.java

```diff
@@ -75,27 +82,49 @@ public void parseCommandScript(File file) throws IOException {
 
     public void parseCommandScript(URL scriptURL) throws IOException {
         try (InputStream scriptStream = scriptURL.openStream()) {
-            BufferedReader br = new BufferedReader(new InputStreamReader(scriptStream));
+            BufferedReader reader = new BufferedReader(new InputStreamReader(scriptStream));
 
-            String commandStr = br.readLine();
-            while (commandStr != null) {                
-                commandStr.trim();
+            String commandStr = reader.readLine();
+            while (commandStr != null) {
+                commandStr = commandStr.trim();
                 // # is a comment
                 if (commandStr.length() > 0 && !commandStr.startsWith("#")) {
+                    commandStr = getEnvironmentSubstitution(commandStr);
                     String command[] = commandStr.split(" ");
                     if (command.length > 1) {
-                        commands.add(new BootCommand(command[0],Arrays.copyOfRange(command, 1, command.length)));
+                        commands.add(new BootCommand(command[0], Arrays.copyOfRange(command, 1, command.length)));
                     } else if (command.length == 1) {
                         commands.add(new BootCommand(command[0]));
                     }
                 }
-                commandStr = br.readLine();
+                commandStr = reader.readLine();
             }
         } catch (IOException ex) {
-            Logger.getLogger(BootCommands.class.getName()).log(Level.SEVERE, null, ex);
+            LOG.log(SEVERE, null, ex);
         }
     }
     
+    private String getEnvironmentSubstitution(String value) {
```

Comment:
This seems to work, but could it not use the `TranslatedConfigView`?

Reply:
Updated the boot commands to use TranslatedConfigView where it requires the pattern ${ENV=key} format for environment variable.